app/front-end/src/features/editor/components/filebarView/filebarGroupItem.tsx

```diff
@@ -26,68 +28,86 @@ import { alpha, Box, IconButton, Typography, useTheme } from '@mui/material';
-        },
-        cursor: blocked ? 'default' : 'pointer',
-        display: 'flex',
-        flexDirection: 'row',
-        alignItems: 'center',
-        gap: '0.5rem',
-      }}
-      onClick={() => {
-        // Update the workspace to the selected file
-        if (!blocked) {
-          Workspace.fileStateUpdate(file);
-          Workspace.filesHistoryStateUpdate(file);
-        }
-      }}
-    >
-      <Typography
+    <>
+      <Box
+        id={id}
         sx={{
-          fontSize: 12,
-          fontWeight: 'bold',
-          textTransform: 'none',
-          maxWidth: '10rem',
-          overflow: 'hidden',
-          textOverflow: 'ellipsis',
-          whiteSpace: 'nowrap',
-        }}
-      >
-        {label}
-      </Typography>
-      <IconButton
-        size='small'
-        disabled={blocked}
-        onClick={(event) => {
-          event.stopPropagation();
+          height: '100%',
+          pl: '1rem',
+          pr: '0.5rem',
+          bgcolor: Workspace.file.id === id ? Theme.palette.background.default : Theme.palette.action.selected,
+          borderRadius: '0rem',
+          ':hover': {
+            backgroundColor:
+              Workspace.file.id === id
+                ? Theme.palette.background.default
+                : blocked
+                  ? Theme.palette.action.selected
+                  : alpha(Theme.palette.background.default, 0.5),
+          },
+          cursor: blocked ? 'default' : 'pointer',
+          display: 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: '0.5rem',
         }}
-        onMouseDown={(event) => {
-          event.stopPropagation();
-          // Remove the file from the workspace
-          Workspace.filesHistoryStateUpdate(undefined, file);
+        onClick={() => {
+          // Update the workspace to the selected file
+          if (!blocked) {
+            if (unsaved) {
+              setIsConfirmDialogOpen(true);
+            } else {
+              Workspace.fileStateUpdate(file);
+              Workspace.filesHistoryStateUpdate(file);
+            }
+          }
         }}
       >
-        <CloseIcon sx={{ fontSize: 12, color: Theme.palette.text.primary }} />
-      </IconButton>
-    </Box>
+        <Typography
+          sx={{
+            fontSize: 12,
+            fontWeight: 'bold',
+            textTransform: 'none',
+            maxWidth: '10rem',
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap',
+          }}
+        >
+          {label}
+        </Typography>
+        <IconButton
+          size='small'
+          disabled={blocked}
+          onClick={(event) => {
+            event.stopPropagation();
+          }}
+          onMouseDown={(event) => {
+            event.stopPropagation();
+            // Remove the file from the workspace
+            Workspace.filesHistoryStateUpdate(undefined, file);
```

Comment:
If closing the currently file which has edited cells, it should also prompt the confirmation about losing changes

Reply:
Resolved! Great catch